service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/TestingSupportControllerTest.java

```diff
@@ -0,0 +1,95 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.web.util.NestedServletException;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+import uk.gov.hmcts.reform.fpl.testingsupport.controllers.TestingSupportController;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(TestingSupportController.class)
+@OverrideAutoConfiguration(enabled = true)
+class TestingSupportControllerTest extends AbstractControllerTest {
+
+    private static final String URL_TEMPLATE = "/testing-support/case/populate/%s";
+    private static final long CASE_ID = 1L;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private CoreCaseDataService coreCaseDataService;
+
+    TestingSupportControllerTest() {
+        super("populate-case");
+    }
+
+    @Test
+    void shouldThrowExceptionForInvalidState() {
+        NestedServletException thrownException = assertThrows(NestedServletException.class,
```

Comment:
you dont have to be tight to specific exception as it is not controlled by us. Exception instead of NestedServletException is fine